### PR TITLE
Monitoring of DNS request time

### PR DESCRIPTION
**Pull request by anderskvist**

### CODE REVIEW

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+func init() {
+	plugins.Register("dnsresponsetime", NewDnsResponseTime)
+}
+
+func NewDnsResponseTime() interface{} {
+	return new(DnsResponseTime)
+}
+
+func (d *DnsResponseTime) Gather(transport plugins.Transport) error {
+
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go]: Remove empty line.

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	domains := strings.Split(d.Domains, ",")
+	servers := strings.Split(d.Servers, ",")
+
+	for _, domain := range domains {
+		for _, server := range servers {
+
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go]: Remove empty line.

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		}
+	}
+
+	return nil
+}
+
+func (d *DnsResponseTime) GetPoints() []*timeseries.Point {
+	points := make([]*timeseries.Point, len(d.Data))
+	for i, data := range d.Data {
+
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go]: Please remove empty line.

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+		}
+
+		values := map[string]interface{}{
+			"loss": data.Loss,
+			"time": data.Time,
+		}
+
+		points[i] = plugins.PointValuesWithTags("dnsresponsetime", values, tags)
+	}
+	return points
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go:79]: Please insert empty line before `return`.

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	Loss   int    `json:"loss"`
+	Time   int64  `json:"time"`
+	Domain string `json:"domain"`
+	Server string `json:"server"`
+}
+
+type DnsResponseTime struct {
+	Data []Data `json:"data"`
+
+	Domains string `toml:"domain" json:"domain" description:"Domain(s) to query"`
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go]: Maybe we should add that these can be separated by commas?

File: plugins/agents/dnsresponsetime/DnsResponseTime.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+		points[i] = plugins.PointValuesWithTags("dnsresponsetime", values, tags)
+	}
+	return points
+}
+
+func (m *DnsResponseTime) GetDoc() *plugins.Doc {
+	doc := plugins.NewDoc("DNS Response Time")
+
+	doc.AddTag("domain", "The domain name to query")
+	doc.AddTag("server", "The server to query")
```
Comment (abrander) [inline plugins/agents/dnsresponsetime/DnsResponseTime.go]: Maybe add a note about how to add multiple servers and domains (eg. commas).

**Review by abrander — CHANGES_REQUESTED** [review]
Thanks for the PR! This looks solid. I have a few comments, mostly about styling and documentation.

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 1 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go]: Done

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 2 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go]: Done

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 3 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go]: Done

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 4 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go:79]: Done

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 5 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go]: Done

File: plugins/agents/dnsresponsetime/DnsResponseTime.go (hunk 6 above)
Comment (anderskvist) [reply plugins/agents/dnsresponsetime/DnsResponseTime.go]: Done

**Review by abrander — APPROVED** [review]
Awesome. Thanks!

**Pull request merged**